File: 2024/day2_part1.py
total = 0

with open("./inputs/day2.txt", "r") as fileData:
  for line in fileData:
    prevStep = 0
    safe = True
    lineData = line.rstrip().split(" ")
    for i in range(0, len(lineData)-1):
      if abs(int(lineData[i]) - int(lineData[i+1])) > 3 or int(lineData[i]) - int(lineData[i+1]) == 0:
        safe = False
        break
      else:
        if prevStep < 0:
          if (int(lineData[i]) - int(lineData[i+1])) * -1 > 0:
            safe = False
            break
        elif prevStep > 0:
          if (int(lineData[i]) - int(lineData[i+1])) * -1 < 0:
            safe = False
            break
        else:
          prevStep = (int(lineData[i]) - int(lineData[i+1])) * -1
    if safe:
      total += 1

print(total)

File: 2025/day2_part1.cs
using System.IO;

string line = File.ReadAllText("./inputs/day2.txt");
string[] ranges = line.Split(",");
long total = 0;

foreach (string range in ranges)
{
  string[] bounds = range.Split("-");
  long lower = long.Parse(bounds[0]);
  long upper = long.Parse(bounds[1]);
  for(long i = lower; i <= upper; i++)
  {
    int numDigits = (int)Math.Floor(Math.Log10(i)) + 1;
    if(numDigits % 2 == 0)
    {
      long part1 = i / (long)Math.Pow(10, numDigits / 2);
      long part2 = i % (long)Math.Pow(10, numDigits / 2);
      if(part1 == part2)
      {
        total += i;
      }
    }
  }
}

Console.WriteLine(total);